app/components/Header/index.js

```diff
@@ -1,32 +1,63 @@
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 import { withRouter } from 'react-router'
+import { Link } from 'react-router-dom'
 
 import { logOutAccount } from '../../actions/account'
 
 class Header extends Component {
-  constructor(props) {
-    super(props)
-  }
 
   handleLogOut() {
     this.props.onLogOut()
 
     this.props.history.push('/create-password')
   }
 
+  handleManageWhitelistedDApps() {
+    this.props.history.push('/whitelisted-dapps')
+  }
+
+  renderMenuButtons() {
+    let url = this.props.location.pathname
+
+    switch (url) {
```

Comment:
Model Component's behaviour based on the url, is not recommended, what happens if you need to have 12 different routes? Create a super big switch loop?
Instead I recommend to take advantage of React's children property:

In account route:
```
<div>
   <Header>
        <div> My custom Header's components :) </div>
   </Header>
   <main>
       ...
   </main>
   ....
</div>
```

And then, in the Header component, do something like:
```
class Header extends React.PureComponent {
    ...
    public render() {
       const { children, ... } = this.props

       return (
          <div>
               <div> Common Header stuff </div>
               { children }  <-- This will render the stuff you created in the account route!! :)
          </div>  
      )
    }
}
```

Do the same for the rest of routes :)